pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -371,10 +374,92 @@ func doUpdate(ctx *zedmanagerContext, uuidStr string,
 	return changed
 }
 
-func doInstall(ctx *zedmanagerContext, uuidStr string,
-	config types.AppInstanceConfig, status *types.AppInstanceStatus) (bool, bool) {
+// doInstallProcessStorageEntriesWithVerifiedImage
+//  Returns: (vs, storageStatusUpdate)
+func doInstallProcessStorageEntriesWithVerifiedImage(
+	ctx *zedmanagerContext,
+	status *types.AppInstanceStatus,
+	ssPtr *types.StorageStatus) (*types.VerifyImageStatus, bool) {
+
+	changed := false
+	safename := ssPtr.Safename()
+
+	// Shortcut if image is already verified
+	vs := lookupVerifyImageStatusAny(ctx, safename, ssPtr.ImageSha256)
+	// Handle post-reboot verification in progress by allowing
+	// types.DOWNLOADED. If the verification later fails we will
+	// get a delete of the VerifyImageStatus and skip this
+	if vs == nil {
+		log.Debugf("Verifier status not found for %s sha %s",
+			safename, ssPtr.ImageSha256)
+		return nil, false
+	}
+	switch vs.State {
+	case types.DELIVERED:
+		log.Infof("Found verified image for %s sha %s\n",
+			safename, ssPtr.ImageSha256)
+
+	case types.DOWNLOADED:
+		log.Infof("Found downloaded/verified image for %s sha %s\n",
+			safename, ssPtr.ImageSha256)
+	default:
+		log.Infof("vs.State (%d) not DELIVERED / DOWNLOADED. safename: %s"+
+			" sha %s", vs.State, safename, ssPtr.ImageSha256)
+		return nil, false
+	}
+	if vs.RefCount == 0 {
+		log.Infof("Vs.RefCount = 0.")
+		// Check if Verifier Config exists.
+		vcPtr := lookupVerifyImageConfig(ctx, safename)
+		if vcPtr == nil {
```

Comment:
Given that handleverifier.go check status.Expired, why not check this here has well?
If Expired is set and there is no VerifyImageConfig, then we know that the verifier has been told it can delete the image.

Reply:
Fixed.. Removed the vcPtr check.. and instead just checked foor Expired state as discussed. With Expired check.. we no longer need the config check.. If Expired. Just trigger re-download.